tmk_core/common/action_util.c

```diff
@@ -130,10 +130,7 @@ void clear_oneshot_layer_state(oneshot_fullfillment_t state)
     oneshot_layer_data &= ~state;
     if (!get_oneshot_layer_state() && start_state != oneshot_layer_data) {
         layer_off(get_oneshot_layer());
-#if (defined(ONESHOT_TIMEOUT) && (ONESHOT_TIMEOUT > 0))
-        oneshot_layer_time = 0;
-#endif
-        oneshot_layers_changed_kb(get_oneshot_layer());
+        reset_oneshot_layer();
```

Comment:
Take special note of this change!

When clearing layer state is reset, the layer is switched off too. This change isn't reflected in the oneshot_layer_data variable, causing my callback to not reflect the current state. I think this is a bug, and so changed the method to call `reset_oneshot_layer` instead.